sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdClientChannelPool.java

```diff
@@ -1184,6 +1190,7 @@ private void notifyChannelConnect(final ChannelFuture future, final Promise<Chan
                     }
 
                     this.safeCloseChannel(channel);
+                    this.connectionStateListener.openConnectionIfNeeded();
```

Comment:
this can be moved into `closeChannel` method - will be easier in the future in case we are calling closeChannel in some other places

Reply:
Resolved.